### PR TITLE
Make Products Changeset Aware 

### PR DESCRIPTION
**Pull request by elisabeth-sorrell**
API PR: https://github.com/SeedCompany/cord-api-v3/pull/2471

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/3466494554) by [Unito](https://www.unito.io)
┆Link To Item: https://seed-company-squad.monday.com/boards/3451697530/pulses/3466494554

### CODE REVIEW

**Comment by elisabeth-sorrell** [comment]
Screen Shot 2022-07-08 at 11 02 29 AM

**Comment by elisabeth-sorrell** [comment]
- Having a little styling problems with the badge outline not stretching to the icon when a prop is changed...
- Not sure if this is part of the scope, but the languageEngagement where the products are added to doesn't have a changeset badge with changed mode.
- Also not sure what to do with the title for a Product with Other type. Should that also have a changeset badge? should it be even changeset aware in the first place?



Also note that I put icons on the right instead of the left. I thought it went better with the flow of how the props are displayed, instead of changing margins on a changeset (the icon overlaps the label when it's on the left.)

File: src/common/fragments/changeset.graphql [hunk 1]
```diff
@@ -31,6 +31,34 @@ fragment ChangesetDiffItem on Resource {
...
+  methodology {
+    value
+  }
+  mediums {
+    value
+  }
+  ...OtherProductProps
+}
+
+fragment OtherProductProps on OtherProduct {
```
Comment (CarsonF) [inline src/common/fragments/changeset.graphql:55]:
```suggestion
  ... on OtherProduct {
```

File: src/components/Changeset/ChangesetBadge.tsx [hunk 2]
```diff
@@ -89,12 +97,15 @@ export const ChangesetBadge = (props: ChangesetBadgeProps) => {
...
+      anchorOrigin={{
+        horizontal: anchorHorizontal ?? 'left',
+        vertical: 'top',
+      }}
       component={BadgeWithTooltip}
       tooltip={(content: ReactElement) => (
         <PaperTooltip
           placement="right"
-          // Assuming more info is given when changed
+          // Assuming more info is given when changedå
```
Comment (CarsonF) [inline src/components/Changeset/ChangesetBadge.tsx:108]:
```suggestion
          // Assuming more info is given when changed
```

File: src/scenes/Products/Create/CreateProduct.tsx [hunk 3]
```diff
@@ -193,8 +205,7 @@ export const CreateProduct = () => {
 
     try {
       await Promise.all([createProduct(), updatePpm()]);
-
-      navigate(`/engagements/${engagementId}`);
+      navigate(`/engagements/${changesetId ? mergedId : engagementId}`);
```
Comment (CarsonF) [inline src/scenes/Products/Create/CreateProduct.tsx:208]:
```suggestion
      navigate(`/engagements/${mergedId}`);
```

File: src/components/Changeset/ChangesetBadge.tsx [hunk 4]
```diff
@@ -71,10 +71,18 @@ export interface ChangesetBadgeProps extends UseStyles<typeof useStyles> {
   disableOutline?: boolean;
   moreInfo?: ReactNode;
   TooltipProps?: Omit<TooltipProps, 'title' | 'children'>;
+  anchorHorizontal?: 'right' | 'left';
```
Comment (CarsonF) [inline src/components/Changeset/ChangesetBadge.tsx:74]: I'd prefer to be more consistent with upstream prop
```suggestion
  anchorOrigin?: Partial<BadgeOrigin>;
```
```ts
anchorOrigin={{ horizontal: 'right' }}
```

**Review by CarsonF — CHANGES_REQUESTED** [review]
Looking good!

Will you rebase this on develop? I think you'll just have some MUI & React imports to adjust.

I need to study the ChangesetPropList component more.

File: src/components/Changeset/ChangesetPropList.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      </List>
+    );
+  }
+  if (!(prop in previous)) {
+    console.error(
+      `${previous.__typename}.${String(
+        prop
+      )} has not been requested in ChangesetDiff`
+    );
+    return;
```
Comment (CarsonF) [inline src/components/Changeset/ChangesetPropList.tsx:64]:
```suggestion
    return null;
```
to make component ReactNode compatible

Comment (elisabeth-sorrell) [reply src/components/Changeset/ChangesetPropList.tsx]: done

**Comment by sethmcknight** [comment]
This lived to the end of its lifetime and died

**Pull request closed without merging**